puffin/src/cli.rs

```diff
@@ -208,6 +225,42 @@ pub fn main<PB: ProtocolBehavior + Clone + 'static>(
         } else {
             return ExitCode::SUCCESS;
         }
+    } else if let Some(matches) = matches.subcommand_matches("execute-traces") {
+        let inputs: ValuesRef<String> = matches.get_many("inputs").unwrap();
+
+        let mut paths = inputs
+            .flat_map(|input| {
+                let input = PathBuf::from(input);
+
+                if input.is_dir() {
+                    fs::read_dir(input)
+                        .expect("failed to read directory")
+                        .map(|entry| entry.expect("failed to read path in directory").path())
+                        .filter(|path| {
+                            !path.file_name().unwrap().to_str().unwrap().starts_with('.')
+                        })
+                        .collect()
+                } else {
+                    vec![input]
+                }
+            })
+            .collect::<Vec<_>>();
+
+        paths.sort_by_key(|path| {
+            fs::metadata(path)
+                .unwrap_or_else(|_| panic!("missing trace file {}", path.display()))
+                .modified()
+                .unwrap()
+        });
```

Comment:
This looks a lot like the code above. Can we deduplicate this?

Reply:
I added this command because I wanted something in CI to run traces and reliably return non-zero exit code when something wrong happened. This code is duplicated on purpose so that in the future we have an incentive to merge these two commands back together.

The full story if you are interested: we had a discussion offline about the current state of the `execute` command. I don't agree with the current use of batching files inside of tlspuffin and I also don't agree with ordering based on files metadata which are unreliable. We agreed to the following:
- we don't want to break the current coverage scripts;
- we need a reliable command exit code for the CI
- we don't want to spend time now to write new coverage scripts and make sure the result is consistent with previous usage.
 
So this is the compromise that makes everyone unhappy: adding a new command that is mostly similar to `execute`. At some point, we will review the way we extract coverage and we can get rid of this duplication.